accounts/services/distributed/__init__.py

```diff
@@ -72,6 +75,36 @@ def delete_session(self, session_id: str) -> None:
         except Exception as e:
             raise SessionDeletionFailed(f'Failed to delete: {e}') from e
 
+    def invalidate_session(self, session_id: str) -> None:
+        """
+        Invalidates a session in the key-value store
+
+        Parameters
+        ----------
+        session_id : str
+        """
+        try:
+            session_data_raw: Optional[Any] = self.get_session(session_id)
+            if session_data_raw is not None:
+                session_data = json.loads(session_data_raw)
+                session_data['end_time'] = time.time()
+                data = json.dumps(session_data)
+                self.r.set(session_id, data)
+            else:
+                raise SessionUnknown(f'Failed to find session {session_id}')
+        except redis.exceptions.ConnectionError as e:
+            raise SessionDeletionFailed(f'Connection failed: {e}') from e
+        except Exception as e:
+            raise SessionDeletionFailed(f'Failed to delete: {e}') from e            
+
+    def get_session(self, id: str) -> Optional[SessionData]: 
+        """Get TapirSession from session id."""
+    
+        try:
+            session = self.r.get(id)
+            return session
+        except Exception as e:
+            return None
```

Comment:
Two thoughts here: 
- I'd suggest not swallowing the exception; might make it more difficult to debug. Instead, consider the pattern we've used for creation and deletion: introduce a ``SessionNotFound`` (or something) exception, and raise it here?
- Consider catching explicit redis exceptions here. What is raised when the key does not exist? That way you can differentiate between cases in which the session really doesn't exists and other kinds of problems (e.g. couldn't connect to Redis, there is a bug in a dependency, etc), and raise/reraise downstream exceptions accordingly.

Reply:
@erickpeirson Seems like the redis exception docs are sparse: https://github.com/andymccurdy/redis-py/blob/master/redis/exceptions.py and the `get` method doesn't explicitly raise exceptions directly, but based on testing, it actually just returns `None`, so I can convert that into an exception. I believe I've now changed the exception handling in invalidate_session and get_session to be what you suggest, which makes sense - can you take a quick peak to see if that is the case?